src/logic/calculate.js

```diff
@@ -0,0 +1,13 @@
+import operate from './operate';
+
+const calculate = (calcDataObj, buttonName) => {
+  if (buttonName === '+/-') {
+    calcDataObj.total *= -1;
+    calcDataObj.next *= -1;
+  } else {
+    calcDataObj.total = operate(calcDataObj.total, calcDataObj.next, buttonName).toString();
+  }
+  return calcDataObj;
+};
```

Comment:
Will this logic be enough? :thinking: 

Remember that this is where all the calculator logic should be. Here is also where a new `state` will be returned.

- You don't want to call the `operate` function when there's only one or no operand, that is `total` and `next` is `null` or `undefined`
- You want to concatenate values here when `buttonName` is a `number` pressed consecutively. E.g. when I press `1` and `3`, I should have `13`.
- You want to handle `=` too

Reply:
I'm not sure I fully understand what should be happening here. I have just added more logic though. Is that what I should have?

Reply:
milestone 5 will tell lol